testgrid/transfigure.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# A script that uses configurator to create a YAML file, and then pushes that
+# YAML file to k8s/test-infra's config/testgrids directory
+#
+# Used by teams that host different instances of Prow, but want to show their
+# results on testgrid.k8s.io
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+main() {
+  branch="transfigure-branch"
+
+  if [[ $# -lt 1 ]]; then
+      echo "Usage: $(basename "$0") </path/to/github/token> [prow_config] [prow-job_config] [testgrid-yaml] [subdir-path]"
+      echo "Path to GitHub token required"
+      exit 1
+  fi
+
+  token=$(readlink -f $1)
+  shift
+  user-from-token
+  ensure-git-config
+  parse-args "$@"
+
+  echo "Getting kubernetes/test-infra repo"
+  git clone https://github.com/kubernetes/test-infra.git || echo "Found test-infra directory; proceeding"
```

Comment:
This conflates clone problems with the repo already existing.

`if [[ ! -d test-infra ]]; then git clone`